### ISSUE TITLE
Discussion: Package Versioning in relation to Angular

### ISSUE THREAD

**Issue by tsvetomir** [telerik/kendo-angular#471]
**TL;DR;**
We propose [dist-tags](https://docs.npmjs.com/cli/dist-tag) as an answer to "Which version of the package should I use if I target Angular X"

---

The packages from the suite follow [semver](http://semver.org/) from the very beginning. This gives you a good idea on what type of changes to expect by looking at the version number.

Currently, we're in the 0.x range:
> Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

With our first official release around the corner, we're about to transition to stable releases:
> Version 1.0.0 defines the public API. The way in which the version number is incremented after this release is dependent on this public API and how it changes.

This versioning would work fine if we only had one version of Angular to target. This is most certainly not the case - Angular 4 will [become a Long Term Support version](https://www.infoq.com/news/2017/04/ng-conf-2017-keynote), with versions 5 and 6 co-existing with it.

Initially we decided that we'd sync our major versions with Angular (see #346). So you know Grid 4 that works with Angular 4, Grid 5 that works with Angular 5 and so on. This convention is very practical, but makes two questionable assumptions:
* Angular will _always_ release major versions on a schedule
* We'll _never_ need to introduce breaking changes between Angular releases

As a solution we propose a simpler scheme that doesn't rely on _"always"_ and _"never"_:
* Each packages increments its versions based strictly on semver
* NPM [dist-tags](https://docs.npmjs.com/cli/dist-tag) such as **ng2** (see #469), **ng5**, etc. point to the latest compatible package
* Omitting the dist-tag will get you the **latest** version, as it currently does

Say if you're targeting Angular 5 you can safely execute:
```
npm install --save @progress/kendo-angular-buttons@ng5 @progress/kendo-angular-l10n@ng5
```

At this point your package.json will contain the major version that the tag resolved to _at this particular moment_. This is good, as `npm update` is guaranteed to pull only compatible versions.

```json
"@progress/kendo-angular-buttons": "^3.1.3",
"@progress/kendo-angular-l10n": "^2.0.1"
```

This allows us to introduce Buttons version 4 that still targets Angular 5 that you can optionally upgrade to. Something that we couldn't do with our versions locked down to Angular.

We'd appreciate your comments.

**Comment by petyosi** [comment]
Question - what does `npm install @progress/foo@ng2` leave in package.json? If the version remains "@ng2" then this seems great.

**Comment by tsvetomir** [comment]
It will resolve to the version that the tag currently points to.
You can opt to change it to `"@progress/foo": "ng2"` at the risk of picking up breaking changes.

**Comment by tsvetomir** [comment]
Closing this as we're out of time. We'll stick with regular semantic versions starting with 1.0
